webapp/src/components/CreateRentalModal/CreateRentalModal.tsx

```diff
@@ -0,0 +1,200 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import {
+  Modal,
+  Button,
+  ModalNavigation,
+  Field,
+  Radio,
+  Header,
+  Popup
+} from 'decentraland-ui'
+import { ContractName, getContract } from 'decentraland-transactions'
+import { TransactionLink } from 'decentraland-dapps/dist/containers'
+import { T, t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { toFixedMANAValue } from 'decentraland-dapps/dist/lib/mana'
+import { PeriodOption } from '../../modules/rental/types'
+import { ManaField } from '../ManaField'
+import { parseMANANumber } from '../../lib/mana'
+import { getDefaultExpirationDate } from '../../modules/order/utils'
+import { Props } from './CreateRentalModal.types'
+import './CreateRentalModal.css'
+import {
+  Authorization,
+  AuthorizationType
+} from 'decentraland-dapps/dist/modules/authorization/types'
+import { hasAuthorization } from 'decentraland-dapps/dist/modules/authorization/utils'
+
+const CreateRentalModal = (props: Props) => {
+  const {
+    open,
+    onCancel,
+    address,
+    isCreating,
+    isGranting,
+    nft,
+    authorizations,
+    onCreate,
+    onGrant,
+    error
+  } = props
+  const [pricePerDayInput, setPricePerDayInput] = useState('')
+  const [periodOptions, setPeriodOptions] = useState<PeriodOption[]>([])
+  const [expiresAt, setExpiresAt] = useState(getDefaultExpirationDate())
+
+  const rentalContractData = getContract(ContractName.Rentals, nft.chainId)
+
+  const authorization: Authorization = useMemo(
+    () => ({
+      address: address!,
+      authorizedAddress: rentalContractData.address,
+      contractAddress: nft.contractAddress,
+      contractName: ContractName.ERC721,
+      chainId: nft.chainId,
+      type: AuthorizationType.APPROVAL
+    }),
+    [address, rentalContractData, nft]
+  )
+
+  const isAuthorized = hasAuthorization(authorizations, authorization)
+
+  const handleSubmit = useCallback(() => {
+    if (!isAuthorized) {
+      onGrant(authorization)
+    } else {
+      onCreate(
+        nft,
+        parseMANANumber(pricePerDayInput),
+        periodOptions,
+        Number(new Date(expiresAt))
+      )
+    }
+  }, [
+    isAuthorized,
+    onCreate,
+    nft,
+    pricePerDayInput,
+    periodOptions,
+    expiresAt,
+    onGrant,
+    authorization
+  ])
+
+  const createOptionHanlder = (periodOption: PeriodOption) => () => {
+    const shouldAdd = !periodOptions.includes(periodOption)
+    if (shouldAdd) {
+      setPeriodOptions([...periodOptions, periodOption])
+    } else {
+      setPeriodOptions(periodOptions.filter(option => option !== periodOption))
+    }
+  }
+
+  // validation
+  const isInvalidPrice = parseMANANumber(pricePerDayInput) <= 0
+  const isInvalidDate = new Date(expiresAt).getTime() < Date.now()
```

Comment:
What do you think about adding `expiration` to this variable name?
```suggestion
  const isInvalidExpirationDate = new Date(expiresAt).getTime() < Date.now()
```

Reply:
done 👍